src/librustc/hir/map/mod.rs

```diff
@@ -931,7 +931,10 @@ impl<'hir> Map<'hir> {
             Some(NodeField(ref f)) => Some(&f.attrs[..]),
             Some(NodeExpr(ref e)) => Some(&*e.attrs),
             Some(NodeStmt(ref s)) => Some(s.node.attrs()),
-            Some(NodeTyParam(tp)) => Some(&tp.attrs[..]),
+            Some(NodeGenericParam(param)) => match param.kind {
+                GenericParamKind::Type { ref attrs, .. } => Some(&attrs[..]),
+                _ => bug!("unexpected non-type NodeGenericParam")
```

Comment:
This seems wrong.

Reply:
There wasn't a `NodeLifetimeParam` (or equivalent) before, but since we removed `TyParam` and subsumed it into `GenericParam`, we need to match on it now. However, it's still only used for the type parameter case at the moment.